src/adapter/src/coord/command_handler.rs

```diff
@@ -549,6 +553,53 @@ impl Coordinator {
                 session,
             ),
 
+            // `CREATE CONNECTION` statements might need validation which happens off the main
+            // coordinator thread of control.
+            stmt @ Statement::CreateConnection(_) => {
+                let plan = match self.plan_statement(&mut session, stmt, &params) {
+                    Ok(Plan::CreateConnection(plan)) => plan,
+                    Ok(_) => unreachable!(),
+                    Err(e) => {
+                        tx.send(Err(e), session);
+                        return;
+                    }
+                };
+
+                if plan.validate {
+                    let internal_cmd_tx = self.internal_cmd_tx.clone();
+                    let conn_id = session.conn_id().clone();
+                    let connection_context = self.connection_context.clone();
+                    let otel_ctx = OpenTelemetryContext::obtain();
+                    task::spawn(|| format!("validate_connection:{conn_id}"), async move {
+                        let connection = &plan.connection.connection;
+                        let result = match connection.validate(&connection_context).await {
+                            Ok(()) => Ok(plan),
+                            Err(err) => Err(err.into()),
+                        };
+
+                        // It is not an error for validation to complete after `internal_cmd_rx` is dropped.
+                        let result =
+                            internal_cmd_tx.send(Message::CreateConnectionValidationReady(
+                                CreateConnectionValidationReady {
+                                    session,
+                                    tx,
+                                    result,
+                                    params,
+                                    depends_on,
+                                    original_stmt,
+                                    otel_ctx,
+                                },
+                            ));
+                        if let Err(e) = result {
+                            tracing::warn!("internal_cmd_rx dropped before we could send: {:?}", e);
+                        }
+                    });
```

Comment:
I don't know if it makes sense to go in the `sequence` stage because this operation is not sequenced until validation happens and other things might get sequenced in between. Additionally, in the event that we cannot actually sequence a `CREATE CONNECTION` after its validation succeeds (it can happen if in the meantime you drop one of its dependencies) then we need access to the original statement so that we can re-plan it. Arguably we should have a mechanism to lock all those in-flight resources from being dropped but we don't have that now

Reply:
I'm going to delegate this part of the review to @mjibson. He's been pulling things off of the main coord thread, so he'd better understand what belongs where.

Reply:
Bumping this comment (@mjibson) I just want to make sure this is consistent with the work you did to move things off the main coord thread.